lib/rules/use-isnan.js

```diff
@@ -184,7 +185,35 @@ module.exports = {
                     node.arguments.length === 1 &&
                     isNaNIdentifier(node.arguments[0])
                 ) {
-                    context.report({ node, messageId: "indexOfNaN", data: { methodName } });
+
+                    /*
+                     * To retain side effects, it's essential to address `NaN` beforehand, which
+                     * is not possible with fixes like `arr.findIndex(Number.isNaN)`.
+                     */
+                    const isSuggestable = node.arguments[0].type !== "SequenceExpression";
+                    const suggestedFixes = [];
+
+                    if (isSuggestable) {
+                        const shouldWrap = callee.property.type === "Literal";
```

Comment:
```suggestion
                        const shouldWrap = callee.computed;
```

Property can also be a template literal, for example::

```js
var firstIndex = myArray[`indexOf`](NaN);
```

Reply:
Nice catch!